module/solutions/helper.js

```diff
@@ -987,17 +991,33 @@ module.exports = class SolutionsHelper {
 						}
 					}
 
-					filterQuery['scope.roles'] = {
-						$in: [CONSTANTS.common.ALL_ROLES, ...data.role.split(',')],
+					if (prefix != '') {
+						filterQuery[`${prefix}.scope.roles`] = {
+							$in: [CONSTANTS.common.ALL_ROLES, ...data.role.split(',')],
+						}
+					} else {
+						filterQuery['scope.roles'] = {
+							$in: [CONSTANTS.common.ALL_ROLES, ...data.role.split(',')],
+						}
 					}
 
 					filterQuery.$or = []
 					Object.keys(_.omit(data, ['filter', 'role', 'factors', 'type'])).forEach((key) => {
-						filterQuery.$or.push({
-							[`scope.${key}`]: { $in: data[key] },
-						})
+						if (prefix != '') {
```

Comment:
@Prajwal17Tunerlabs can we add some comments here so that we will get an understanding how the fetching is happening

Reply:
comments added to the code.